rcl/src/rcl/wait.c

```diff
@@ -64,6 +64,7 @@ static void
 __wait_set_clean_up(rcl_wait_set_t * wait_set, rcl_allocator_t allocator)
 {
   rcl_ret_t ret;
+  (void)ret;  // NO LINT
```

Comment:
Well, I interpret that guideline as particularly for a scenario like this:

```c++
// This way:
int foo()
{
  {
    int i;
    i  = 3;
  }
}
// Rather than this way:
int foo()
{
  int i;
  {
    i = 3;
  }
}
```

But in this scenario we're actually reusing the variable across multiple scopes. But since we're not getting any advantage with this (other than less code) I changed it in 3310359.